r/R/array.R

```diff
@@ -119,9 +119,9 @@ Array <- R6Class("Array",
     },
     Slice = function(offset, length = NULL) {
       if (is.null(length)) {
-        Array$create(Array__Slice1(self, offset))
+        Array$create(Array__Slice1(self, as_index(offset)))
       } else {
-        Array$create(Array__Slice2(self, offset, length))
+        Array$create(Array__Slice2(self, as_index(offset), length))
```

Comment:
If `offset` needs this then so should `length`. Also I expect that you'll want/have to use this in lots of other places.

```suggestion
        Array$create(Array__Slice2(self, as_index(offset), as_index(length)))
```

Reply:
Yeah, I'm wondering if this could be handled by a C++ class instead. For now, I'm just trying to make it all pass. This is only needed if converting to `int` refuses to deal with `NA` (which is a logical), i.e. https://github.com/r-lib/cpp11/pull/53